platform/darwin/src/MGLCircleStyleLayer.mm

```diff
@@ -47,6 +47,14 @@ - (instancetype)initWithIdentifier:(NSString *)identifier source:(MGLSource *)so
     return self;
 }
 
+- (instancetype)initWithRawLayer:(mbgl::style::CircleLayer *)rawLayer source:(MGLSource *)source
+{
+    if (self = [super initWithIdentifier:@(rawLayer->getID().c_str()) source:source]) {
```

Comment:
Not sure I follow. Are you suggesting to remove the `source` parameter from the various `initWithRawLayer`s? How would the implementation of `initWithRawLayer` obtain a `mbgl::style::Source *` to pass to `+[MGLSource sourceWithRawSource:]`?